cmd/kubeapps-apis/server/packages.go

```diff
@@ -68,3 +68,41 @@ func (s packagesServer) GetAvailablePackageSummaries(ctx context.Context, reques
 		AvailablePackagesSummaries: pkgs,
 	}, nil
 }
+
+// GetAvailablePackages returns the packages based on the request.
+func (s packagesServer) GetAvailablePackageDetail(ctx context.Context, request *packages.GetAvailablePackageDetailRequest) (*packages.GetAvailablePackageDetailResponse, error) {
+	contextMsg := ""
+	if request.AvailablePackageRef != nil && request.AvailablePackageRef.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.AvailablePackageRef.Context.Cluster, request.AvailablePackageRef.Context.Namespace)
+	}
+
+	log.Infof("+core GetAvailablePackageDetail %s", contextMsg)
+
+	pkg := &packages.AvailablePackageDetail{}
+
+	// TODO: We can do these in parallel in separate go routines.
+	for _, p := range s.plugins {
+
+		response, err := p.server.GetAvailablePackageDetail(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		if response.AvailablePackageDetail != nil {
+			pkg = response.AvailablePackageDetail
+
+			if response.AvailablePackageDetail.AvailablePackageRef == nil {
+				pkg.AvailablePackageRef = &packages.AvailablePackageReference{}
+			}
+			pkg.AvailablePackageRef.Plugin = p.plugin
```

Comment:
This should no longer be necessary. After discussion, we've updated to always filling in the plugin in the reference. If you check `GetAvailablePackageDetail` for the helm plugin, you'll see it's already including it.